balena/build.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+echo "TASKINFO: Build the application in BalenaCloud"
+set -e
+SRC_PATH="$1"
+set -u
+
+test "${DEBUG}" == "false" || set -x
+
+balena_app_exists() {
+  return $(balena apps | grep -q "$1")
+}
+
+get_project_name_from_source() {
+    echo "$(git -C $SRC_PATH remote -v | head -1 | sed 's|.*/\(.*\)\.git.*$|\1|g')"
+}
+
+app_name=$(get_project_name_from_source)
+
+balena login -t $API_KEY
+if ! balena_app_exists "$app_name"; then
+  balena app create $app_name -t $DEVICE_TYPE
```

Comment:
Do we have any guarantee from the concourse side this will not run in parallel for the same project?
Otherwise, there will be a collision.

Reply:
At this stage there's no parallelization and therefore the jobs are run sequentially :)

Reply:
@kaisoz that is not correct, multiple PRs can be triggered at the same time and run in parallel. I think in this case its not a big deal, as in most cases one will fail on the creation and the PR will simply have to be re-tested

Reply:
Ahh true @nazrhom, I misunderstood it. I just tried two run two `balena app create` and yes, one of them fails with the `Unique key constraint violated` error. This can be solved if we check if the output of the `balena app create` contains such string and in that case continue the procedure, but in that case we'll need to disable `set -e` since the command will return 1. I can create a to-be-applied-later PR fixing this :)